File: ProgressSnackbar/app/src/main/java/prateek_gupta/progresssnackbar/MainActivity.java
package prateek_gupta.progresssnackbar;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.ProgressBar;
import android.widget.TextView;

import com.google.android.material.snackbar.Snackbar;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Creating a SnackBar
        Snackbar snackBar = Snackbar.make(findViewById(R.id.main_view),
                "Something", Snackbar.LENGTH_INDEFINITE);

        // Retrieving SnackbarLayout for the snackBar
        Snackbar.SnackbarLayout snackbarLayout =
                (Snackbar.SnackbarLayout) snackBar.getView();

        // Retrieving the default TextView of the SnackBar
        TextView textView = (TextView) snackbarLayout.findViewById(
                com.google.android.material.R.id.snackbar_text);

        // Using INVISIBLE not GONE, to keep the basic structure of the SnackBar
        textView.setVisibility(View.INVISIBLE);

        // Adding our custom view to snackBar
        snackbarLayout.addView(getLayoutInflater().inflate(R.layout.progress_snackbar,
                null));

        // Displaying snackBar
        snackBar.show();
    }
}

File: ProgressSnackbar/app/src/main/java/prateek_gupta/progresssnackbar/MainActivity.kt
package prateek_gupta.progresssnackbar

import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.snackbar.Snackbar
import com.google.android.material.snackbar.Snackbar.SnackbarLayout

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Creating a SnackBar
        val snackBar = Snackbar.make(
            findViewById(R.id.main_view),
            "Something", Snackbar.LENGTH_INDEFINITE
        )

        // Retrieving SnackbarLayout for the snackBar
        val snackbarLayout = snackBar.view as SnackbarLayout

        // Retrieving the default TextView of the SnackBar
        val textView = snackbarLayout.findViewById<View>(
            com.google.android.material.R.id.snackbar_text
        ) as TextView

        // Using INVISIBLE not GONE, to keep the basic structure of the SnackBar
        textView.visibility = View.INVISIBLE

        // Adding our custom view to snackBar
        snackbarLayout.addView(
            layoutInflater.inflate(
                R.layout.progress_snackbar,
                null
            )
        )

        // Displaying snackBar
        snackBar.show()
    }
}